tests/datasets/test_datadoc_metadata.py

```diff
@@ -904,3 +904,48 @@ def test_check_ready_to_merge_inconsistent_variable_data_types(
             ],
             errors_as_warnings=errors_as_warnings,
         )
+
+
+@pytest.mark.parametrize(
+    "existing_metadata_path",
+    [TEST_EXISTING_METADATA_DIRECTORY / "dataset_and_pseudo"],
+)
+def test_existing_metadata_file_with_pseudonymization(
+    existing_metadata_file: Path,  # noqa: ARG001
+    metadata: Datadoc,
+):
+    assert len(metadata.pseudo_variables) == 2
+    metadata.add_pseudo_variable("sykepenger")
+    assert len(metadata.pseudo_variables) == 3
+
+
+@pytest.mark.parametrize(
+    "existing_metadata_path",
+    [TEST_EXISTING_METADATA_DIRECTORY / "dataset_and_pseudo"],
+)
+def test_existing_metadata_file_with_pseudonymization_raises(
+    existing_metadata_file: Path,  # noqa: ARG001
+    metadata: Datadoc,
+):
+    with pytest.raises(KeyError):  # replace with the actual expected exception
+        metadata.add_pseudo_variable("new_pseudo_variable")
```

Comment:
Does this need to be updated as per the comment? Is the reason we throw an exception because `"new_pseudo_variable"` doesn't exist in the variables list? If so it would be good to describe that in the test name.